src/main/perl/FileWriter.pm

```diff
@@ -173,8 +172,8 @@ sub close
                                   keeps_state => 1);
         $cmd->execute();
         *$self->{LOG}->verbose ("Changes to ", *$self->{filename}, ":");
+        $diff = "" if (! defined($diff));
```

Comment:
this change is to get rid of some 'undefined' warnings (and i think only in the unittests). there's also https://github.com/quattor/CAF/pull/119 to handle empty diffs